Jenkinsfile_k8s

```diff
@@ -1,7 +1,11 @@
-buildDockerAndPublishImage('jenkins-weekly', [
-    mainBranch: 'main',
-    automaticSemanticVersioning: true,
-    gitCredentials: 'github-app-infra',
-    nextVersionCommand: 'jx-release-version -next-version=semantic:strip-prerelease',
-    metadataFromSh: 'cat Dockerfile | grep "FROM jenkins" | sed "s|FROM jenkins/jenkins:|-|" | sed "s|-jdk11||"'
-])
+retry(3) {
```

Comment:
could also move this down into the library making retry and timeout configurable with sensible defaults?

Reply:
Good question. The reasons why I did not are:
- I want to try on a small area first, to see if it has an impact (so area is this repository for now)
- Adding such a behavior in the pipeline-library would be a dedicated function, not inside the existing one as it need to be reused and because it would bloat the existing feature. But you're making a valid point to make it reusable as we need this "retry/timeout" logic on a lot of places, starting with release processes